exp/controllers/helpers.go

```diff
@@ -59,7 +59,7 @@ func AzureClusterToAzureMachinePoolsMapper(ctx context.Context, c client.Client,
 
 		azCluster, ok := o.(*infrav1.AzureCluster)
 		if !ok {
-			log.Error(errors.Errorf("expected an AzureCluster, got %T instead", o.GetObjectKind()), "failed to map AzureCluster")
+			log.Error(errors.Errorf("expected an AzureCluster, got %T instead", o), "failed to map AzureCluster")
```

Comment:
Before, the log looked like this, regardless of the underlying type of `o`:
```
expected an AzureCluster, got *v1.TypeMeta instead
```

Now it could look something like
```
expected an AzureCluster, got *v1beta1.AzureManagedCluster instead
```